src/byte/writer.rs

```diff
@@ -152,11 +158,28 @@ impl ByteWriter {
         }
     }
 
+    /// Write data asynchronously.
+    ///
+    /// # Examples
+    /// ```ignore
+    /// let mut byte_writer = client_factory.create_byte_writer_async(segment).await;
+    /// let payload = vec![0; 8];
+    /// let (size, handle) = byte_writer.write(&payload).await;
+    /// assert!(handle.await.is_ok());
+    /// ```
+    pub async fn write_async(&mut self, buf: &[u8]) -> (usize, oneshot::Receiver<Result<(), Error>>) {
```

Comment:
The default path is already buffered and we don't have acks. So what if, rather than modeling it on the event path which has oneshots, and the caller does `.await.await`, we just have `write_async` and `flush_async` which are both identical to `write` and `flush` except they don't block. This avoid the need for the oneshots.